From jhovold@gmail.com  Fri Apr 23 15:35:42 2010
From: Johan Hovold <jhovold@gmail.com>
Date: Wed, 17 Mar 2010 23:06:06 +0100
Subject: USB: serial: allow custom multi-urb write bulk callbacks
To: linux-usb@vger.kernel.org
Cc: Greg Kroah-Hartman <gregkh@suse.de>, Johan Hovold <jhovold@gmail.com>
Message-ID: <1268863572-17891-15-git-send-email-jhovold@gmail.com>


Allow drivers to implement their own multi-urb write bulk callbacks as
we do for single urb writes.

Signed-off-by: Johan Hovold <jhovold@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/generic.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/generic.c
+++ b/drivers/usb/serial/generic.c
@@ -217,7 +217,7 @@ static int usb_serial_multi_urb_write(st
 			usb_sndbulkpipe(port->serial->dev,
 					port->bulk_out_endpointAddress),
 			buffer, towrite,
-			usb_serial_generic_write_bulk_callback, port);
+			port->serial->type->write_bulk_callback, port);
 
 		status = usb_submit_urb(urb, GFP_ATOMIC);
 		if (status) {
